### ISSUE TITLE
Failed calling webhook "webhook.openpolicyagent.org" error

### ISSUE THREAD

**Issue by marcusdroy** [open-policy-agent/kube-mgmt#197]
While validating upgrading to the latest version of OPA and kube-mgmt on Kubernetes 1.26, we are seeing `Failed calling webhook "webhook.openpolicyagent.org" error` in the api server logs.

E0321 15:06:15.586026  dispatcher.go:183] failed calling webhook "webhook.openpolicyagent.org": failed to call webhook: the server could not find the requested resource
W0321 15:06:17.357212 1 dispatcher.go:176] Failed calling webhook, failing open webhook.openpolicyagent.org: failed calling webhook "webhook.openpolicyagent.org": failed to call webhook: the server could not find the requested resource
E0321 15:06:17.357275  1 dispatcher.go:183] failed calling webhook "webhook.openpolicyagent.org": failed to call webhook: the server could not find the requested resource

**Reproduction Steps**
Starting with blank kind cluster `kind create cluster --image=kindest/node:v1.26.0`

1. Install latest chart via Helm
`helm upgrade -i -n opa --create-namespace opa opa/opa-kube-mgmt --set admissionController.enabled=true --version 8.1.0`

2. validatingwebhookconfiguration is created as expected below

```yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  annotations:
    meta.helm.sh/release-name: opa
    meta.helm.sh/release-namespace: opa
  creationTimestamp: "2023-03-21T15:04:27Z"
  generation: 1
  labels:
    app: opa-opa-kube-mgmt
    app.kubernetes.io/managed-by: Helm
    chart: opa-kube-mgmt-8.1.0
    heritage: Helm
    release: opa
  name: opa-opa-kube-mgmt
  resourceVersion: "622"
  uid: c696feb5-c5dc-4fbd-a842-c02105027269
webhooks:
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    caBundle: "<redacted>"
    service:
      name: opa-opa-kube-mgmt
      namespace: opa
      port: 8181
  failurePolicy: Ignore
  matchPolicy: Equivalent
  name: webhook.openpolicyagent.org
  namespaceSelector:
    matchExpressions:
    - key: openpolicyagent.org/webhook
      operator: NotIn
      values:
      - ignore
  objectSelector: {}
  rules:
  - apiGroups:
    - '*'
    apiVersions:
    - '*'
    operations:
    - '*'
    resources:
    - '*'
    scope: '*'
  sideEffects: None
  timeoutSeconds: 10
```
  
  3. Service is available `k get svc opa-opa-kube-mgmt -n opa`
 
  `NAME                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
opa-opa-kube-mgmt   ClusterIP   10.96.164.211   <none>        8181/TCP   5m38s
`

5. However seeing this error in the api server logs
`
E0321 15:06:15.586026 dispatcher.go:183] failed calling webhook "webhook.openpolicyagent.org": failed to call webhook: the server could not find the requested resource
W0321 15:06:17.357212 1 dispatcher.go:176] Failed calling webhook, failing open webhook.openpolicyagent.org: failed calling webhook "webhook.openpolicyagent.org": failed to call webhook: the server could not find the requested resource
E0321 15:06:17.357275 1 dispatcher.go:183] failed calling webhook "webhook.openpolicyagent.org": failed to call webhook: the server could not find the requested resource
`

**Comment by charlieegan3** [comment]
Yeah, that doesn't seem right. I  think that the url is going to need to be set in the client config too, i.e. https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#url

I think that you want to set this to this path: https://www.openpolicyagent.org/docs/latest/rest-api/#get-a-document-webhook i.e. `/v0/data/system/main` if you have the same structure of policy here: https://www.openpolicyagent.org/docs/latest/kubernetes-tutorial/#policy-2-prohibit-hostname-conflicts

**Comment by marcusdroy** [comment]
I tried the configuration below but got a auth error:

```
    service:
      name: opa-opa-kube-mgmt
      namespace: opa
      path: /v0/data/system/main
      port: 8181
```

`
failed calling webhook "webhook.openpolicyagent.org": failed to call webhook: the server has asked for the client to provide credentials` 

In the opa pod logs it has:

`{"client_addr":"10.244.0.1:48895","level":"info","msg":"Sent response.","req_id":11321,"req_method":"POST","req_path":"/v0/data/system/main","resp_bytes":87,"resp_duration":0.884191,"resp_status":401,"time":"2023-03-21T17:25:42Z"}`

I havent loaded any bundles in yet besides what the chart is providing
